pennylane/fermi/fermionic.py

```diff
@@ -299,9 +299,13 @@ def to_mat(self, n_orbitals=None):
             )
 
         largest_order = n_orbitals or largest_orb_id
-        mat = qml.jordan_wigner(self, ps=True).to_mat(wire_order=list(range(largest_order)))
 
-        return mat
+        if format == "dense":
+            return qml.jordan_wigner(self, ps=True).to_mat(wire_order=list(range(largest_order)))
+
+        return qml.jordan_wigner(self, ps=True).to_mat(
```

Comment:
Why cannot we simply do `qml.jordan_wigner(self, ps=True).to_mat(wire_order=soemthing, format=format)`?